contracts/src/ERC20Destination.sol

```diff
@@ -92,14 +108,55 @@ contract ERC20Destination is IERC20Bridge, TeleporterTokenDestination, ERC20 {
      *
      * Calls {ERC20-_burn} to burn tokens from this contract.
      */
-    function _burn(uint256 amount) internal virtual override {
+    function _burn(uint256 amount) internal override {
         _burn(address(this), amount);
     }
 
     /**
-     * @dev See {TeleporterTokenDestination-_scaleTokens}
+     * @dev See {TeleporterTokenDestination-_handleSendAndCall}
+     *
+     * Mints the tokens to this contract, approves the recipient contract to spend them,
+     * and calls {IERC20SendAndCallReceiver-receiveTokens} on the recipient contract.
+     * If the call fails or doesn't spend all of the tokens, the remaining amount is
+     * sent to the fallback recipient.
      */
-    function _scaleTokens(uint256 value, bool) internal pure override returns (uint256) {
-        return value;
+    function _handleSendAndCall(
+        SingleHopCallMessage memory message,
+        uint256 amount
+    ) internal override {
+        // Mint the tokens to this contract address.
+        _mint(address(this), amount);
+
+        // Approve the destination contract to spend the amount.
+        _approve(address(this), message.recipientContract, amount);
+
+        // Encode the call to {IERC20SendAndCallReceiver-receiveTokens}
+        bytes memory payload = abi.encodeCall(
+            IERC20SendAndCallReceiver.receiveTokens,
+            (address(this), amount, message.recipientPayload)
+        );
+
+        // Call the destination contract with the given payload and gas amount.
+        bool success = CallUtils._callWithExactGas(
+            message.recipientGasLimit, message.recipientContract, payload
+        );
+
+        // Check what the remaining allowance is to transfer to the fallback recipient.
+        uint256 remainingAllowance = allowance(address(this), message.recipientContract);
+
+        // Reset the destination contract allowance to 0.
+        _approve(address(this), message.recipientContract, 0);
+
+        if (success) {
+            emit CallSucceeded(message.recipientContract, amount);
+        } else {
+            emit CallFailed(message.recipientContract, amount);
+        }
+
+        // Transfer any remaining allowance to the fallback recipient. This will be the
+        // full amount if the call failed.
```

Comment:
That's correct, though really even if it could, we check the remaining allowance amount left unspent whether or not the call succeed, so it should still be correct.